### PR TITLE
backend for in xor filter work

### PR DESCRIPTION
**Pull request by ambuznego**
`in_xor_filter` column to be used to display when device has NOT YET entered XOR filter

### CODE REVIEW

File: lib/console_web/controllers/router/device_controller.ex [hunk 1]
```diff
@@ -425,4 +425,24 @@ defmodule ConsoleWeb.Router.DeviceController do
...
   end
+
+  def update_devices_in_xor_filter(conn, %{"added" => added_device_ids, "removed" => removed_device_ids}) do
+    result = Ecto.Multi.new()
+      |> Ecto.Multi.run(:added_devices, fn _repo, _ ->
+        with {:ok, devices} <- Devices.update_in_xor_filter(added_device_ids, true) do
+          {:ok, devices}
+        end
+      end)
+      |> Ecto.Multi.run(:removed_devices, fn _repo, _ ->
```
Comment (vicmgs) [inline lib/console_web/controllers/router/device_controller.ex:436]: I think we dont need to remove any devices. Michael is just sending them to cover all bases

File: lib/console/devices/devices.ex [hunk 2]
```diff
@@ -172,4 +172,10 @@ defmodule Console.Devices do
     devices = from(d in Device, where: d.organization_id == ^organization.id) |> Repo.all()
     length(devices)
   end
+
+  def update_in_xor_filter(device_ids, value) do
+    from(d in Device, where: d.id in ^device_ids) |> Repo.update_all(set: [in_xor_filter: value])
```
Comment (vicmgs) [inline lib/console/devices/devices.ex:177]: Given how we use this function in the controller file, it doesnt seem necessary to do the next line. We can just return the result of Repo.update_all

Comment (ambuznego) [reply lib/console/devices/devices.ex:177]: you is right

File: lib/console_web/controllers/router/device_controller.ex (hunk 1 above)
Comment (ambuznego) [reply lib/console_web/controllers/router/device_controller.ex:436]: oh gotcha. will remove

**Review by vicmgs — APPROVED** [review]

**Pull request merged**